File: flutter/test_app/lib/data/supabase/repository/auth_repository_impl.dart

import '../../../domain/model/user_role.dart';
import '../../../domain/repository/auth_repository.dart';
import '../network/service/supabase_auth_service.dart';

class AuthRepositoryImpl implements AuthRepository {
  final SupabaseAuthService supabaseAuthService;

  AuthRepositoryImpl(this.supabaseAuthService);

  @override
  Future<void> signIn(String email, String password) async {
    await supabaseAuthService.signIn(
      email: email,
      password: password,
    );
  }

  @override
  Future<void> signOut() async {
    await supabaseAuthService.signOut();
  }

  @override
  String? currentUserId() {
    return supabaseAuthService.currentUserId();
  }

  @override
  Future<UserRole> getUserRole() async {
    return await supabaseAuthService.getCurrentUserRole();
  }
}

File: kotlin_jetpack_compose/app/src/main/java/com/example/testapp/data/supabase/repository/AuthRepositoryImpl.kt
package com.example.testapp.data.supabase.repository

import com.example.testapp.data.supabase.network.service.SupabaseAuthService
import com.example.testapp.domain.model.UserRole
import com.example.testapp.domain.repository.AuthRepository
import io.github.jan.supabase.SupabaseClient
import io.github.jan.supabase.auth.auth
import javax.inject.Inject
import javax.inject.Singleton
import io.github.jan.supabase.auth.providers.builtin.Email

@Singleton
class AuthRepositoryImpl @Inject constructor(
    private val supabaseAuthService: SupabaseAuthService
) : AuthRepository {

    override suspend fun signIn(email: String, password: String) {
        supabaseAuthService.signIn(email, password)
    }

    override suspend fun signOut() {
        supabaseAuthService.signOut()
    }

    override fun currentUserId(): String? {
        return supabaseAuthService.currentUserId()
    }

    override suspend fun getUserRole(): UserRole {
        return supabaseAuthService.getCurrentUserRole()
    }
}